Client/Frontend/Settings/SearchSettingsTableViewController.swift

```diff
@@ -312,23 +312,47 @@ class SearchSettingsTableViewController: UITableViewController {
   override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
     if editingStyle == .delete {
       guard let engine = customSearchEngines[safe: indexPath.row] else { return }
+      
+      func deleteCustomEngine() {
+        do {
+          try searchEngines.deleteCustomEngine(engine)
+          tableView.deleteRows(at: [indexPath], with: .right)
+          tableView.reloadData()
```

Comment:
I don't think you have to do both `deleteRows` and `reloadData` it's usually one or the other.

Reply:
Not in this case, I want that row to be deleted with animation and the table should be updated to reflect the default search engine change.